qiskit/quantum_info/operators/symplectic/clifford.py

```diff
@@ -0,0 +1,526 @@
+        else:
+            self._table = StabilizerTable(data)
+
+            # Validate table is a symplectic matrix
+            if validate and not Clifford._is_symplectic(self._table.array):
+                raise QiskitError(
+                    'Invalid Clifford. Input StabilizerTable is not a valid'
+                    ' symplectic matrix.')
+
+        # Initialize BaseOperator
+        dims = self._table.num_qubits * (2,)
+        super().__init__(dims, dims)
+
+    def __repr__(self):
+        return 'Clifford({})'.format(repr(self.table))
+
+    def __str__(self):
+        return 'Clifford: Stabilizer = {}, Destabilizer = {}'.format(
+            str(self.stabilizer.to_labels()),
+            str(self.destabilizer.to_labels()))
+
+    def __eq__(self, other):
+        """Check if two Clifford tables are equal"""
+        return super().__eq__(other) and self._table == other._table
+
+    # ---------------------------------------------------------------------
+    # Attributes
+    # ---------------------------------------------------------------------
+    def __getitem__(self, key):
+        """Return a stabilizer Pauli row"""
+        return self._table.__getitem__(key)
+
+    def __setitem__(self, key, value):
+        """Set a stabilizer Pauli row"""
+        self._table.__setitem__(key, value)
+
+    @property
+    def table(self):
+        """Return StabilizerTable"""
+        return self._table
+
+    @table.setter
+    def table(self, value):
+        """Set the stabilizer table"""
+        # Note this setter cannot change the size of the Clifford
+        # It can only replace the contents of the StabilizerTable with
+        # another StabilizerTable of the same size.
+        if not isinstance(value, StabilizerTable):
+            value = StabilizerTable(value)
+        self._table._array[:, :] = value._table._array
+        self._table._phase[:] = value._table._phase
+
+    @property
+    def stabilizer(self):
+        """Return the stabilizer block of the StabilizerTable."""
+        return StabilizerTable(self._table[self.num_qubits:2*self.num_qubits])
+
+    @stabilizer.setter
+    def stabilizer(self, value):
+        """Set the value of stabilizer block of the StabilizerTable"""
+        inds = slice(self.num_qubits, 2*self.num_qubits)
+        self._table.__setitem__(inds, value)
+
+    @property
+    def destabilizer(self):
+        """Return the destabilizer block of the StabilizerTable."""
+        return StabilizerTable(self._table[0:self.num_qubits])
+
+    @destabilizer.setter
+    def destabilizer(self, value):
+        """Set the value of destabilizer block of the StabilizerTable"""
+        inds = slice(0, self.num_qubits)
+        self._table.__setitem__(inds, value)
+
+    # ---------------------------------------------------------------------
+    # Utility Operator methods
+    # ---------------------------------------------------------------------
+
+    def is_unitary(self, atol=None, rtol=None):
```

Comment:
why are the parameters atol and rtol are needed here?

Reply:
They aren't, this used to be an abstract method of base class but has since been removed.